crates/noirc_frontend/src/hir/type_check/errors.rs

```diff
@@ -96,6 +98,11 @@ impl From<TypeCheckError> for Diagnostic {
                 format!("return type is {typ}"),
                 span,
             ),
+            TypeCheckError::TypeAnnotationsNeeded { span } => Diagnostic::simple_error(
+                "Type annotations needed before this point.".to_string(),
+                "Try adding a type annotation for this value before it is used here".to_string(),
```

Comment:
I think the rust message is clearer and would prefer that we use it instead:
Replace "Try adding a type annotation for this value before it is used here"
with: "type must be known at this point"

Reply:
This is a bit repetitive with the first line of the message so I've updated to change both lines